modules/pool/lib/apr-data-sources/ib-yield-apr-handlers/sources/aave-apr-handler.ts

```diff
@@ -0,0 +1,232 @@
+import axios from 'axios';
+import { AprHandler } from '../ib-yield-apr-handlers';
+
+class AaveAprHandler implements AprHandler {
+    wrappedTokens: { [key: string]: string };
+    aaveTokens: { [key: string]: string };
+    underlyingTokens: { [key: string]: string };
+    subgraphUrl: string;
+    networkPrismaId: string;
+
+    readonly group = 'AAVE';
+
+    readonly query = `query getReserves($aTokens: [String!], $underlyingAssets: [Bytes!]) {
+    reserves(
+      where: {
+        aToken_in: $aTokens
+        underlyingAsset_in: $underlyingAssets
+        isActive: true
+      }
+    ) {
+      id
+      underlyingAsset
+      liquidityRate
+    }
+  }`;
+
+    constructor(aprHandlerConfig: AaveAprHandlerConfig) {
+        this.wrappedTokens = aprHandlerConfig.wrappedTokens;
+        this.aaveTokens = aprHandlerConfig.aaveTokens;
+        this.underlyingTokens = aprHandlerConfig.underlyingTokens;
+        this.subgraphUrl = aprHandlerConfig.subgraphUrl;
+        this.networkPrismaId = aprHandlerConfig.network;
+    }
+
+    async getAprs() {
+        try {
+            const requestQuery = {
+                operationName: 'getReserves',
+                query: this.query,
+                variables: {
+                    aTokens: Object.values(this.aaveTokens),
+                    underlyingAssets: Object.values(this.underlyingTokens),
+                },
+            };
+            const { data } = await axios({
+                url: this.subgraphUrl,
+                method: 'post',
+                data: requestQuery,
+                headers: { 'Content-Type': 'application/json' },
+            });
+            const {
+                data: { reserves },
+            } = data as ReserveResponse;
+
+            const aprsByUnderlyingAddress = Object.fromEntries(
+                reserves.map((r) => [
+                    r.underlyingAsset,
+                    // Converting from aave ray number (27 digits) to float
+                    Number(r.liquidityRate.slice(0, 27)) / 1e27,
+                ]),
+            );
+            const aprEntries = Object.fromEntries(
+                Object.values(this.underlyingTokens)
+                    //Removing undefined aprs
+                    .filter(([, address]) => !!aprsByUnderlyingAddress[address])
+                    //Mapping aprs by wrapped instead of underlying addresses
+                    .map(([underlyingTokenName, underlyingTokenAddress]) => [
+                        this.wrappedTokens['wa' + underlyingTokenName] as string,
+                        aprsByUnderlyingAddress[underlyingTokenAddress],
+                    ]),
+            );
+            return aprEntries;
+        } catch (e) {
+            console.error(`Failed to fetch Aave APR in subgraph ${this.subgraphUrl}:`, e);
+            return {};
+        }
+    }
+}
+
+const wrappedAaveTokensV2Mainnet = {
```

Comment:
I think all of this config makes more sense in the network config files. Or something like:

```
aaveTokens: {
  [Network.Mainnet]: {
    tokensV2: {
      ...
    },
    wrappedTokensV2: {
      ...
    }, 
    tokensV3: {
      ...
    }
 }
}
```

(The underlying tokens should be defined somewhere else as they are not Aave specific)

Then having one function `getAaveAprHander(network: Network, version: number)` that pulls the correct config settings and returns a new `AaveAprHandler` with it, instead of defining a function for every network. 

In general network names should never need to be added to variables, it's better to use a map. 

This will make it much easier to extend as we add more networks, as eventually we may have 50+ networks and this will become messy.